code/datums/diseases/advance/symptoms/organs.dm

```diff
@@ -0,0 +1,226 @@
+		return
+	switch(A.stage)
+		if(4, 5)
+			M.restoreEars()
+
+			if(HAS_TRAIT_FROM(M, TRAIT_BLIND, EYE_DAMAGE))
+				if(prob(20))
+					to_chat(M, "<span class='notice'>Your vision slowly returns...</span>")
+					M.cure_blind(EYE_DAMAGE)
+					M.cure_nearsighted(EYE_DAMAGE)
+					M.blur_eyes(35)
+
+				else if(HAS_TRAIT_FROM(M, TRAIT_NEARSIGHT, EYE_DAMAGE))
+					to_chat(M, "<span class='notice'>You can finally focus your eyes on distant objects.</span>")
+					M.cure_nearsighted(EYE_DAMAGE)
+					M.blur_eyes(10)
+
+				else if(M.eye_blind || M.eye_blurry)
+					M.set_blindness(0)
+					M.set_blurriness(0)
+				else if(eyes.damage > 0)
+					eyes.applyOrganDamage(-1)
+		else
+			if(prob(base_message_chance))
+				to_chat(M, "<span class='notice'>[pick("Your eyes feel great.","You feel like your eyes can focus more clearly.", "You don't feel the need to blink.","Your ears feel great.","Your healing feels more acute.")]</span>")
+
+
+/datum/symptom/organ_restoration //heals damage to other internal organs that get damaged far less often
+	name = "Organ Restoration"
+	desc = "The virus stimulates rapid cell growth in organ tissues, slowly repairing the host's organs over time."
+	stealth = 2
+	resistance = 3
+	stage_speed = -2
+	transmission = -1
+	level = 6
+	severity = -1
+	symptom_delay_min = 1
+	symptom_delay_max = 1
+	prefixes = list("Organ 9")
+	var/curing = FALSE
+	var/regenorgans = FALSE 
+	threshold_desc = "<b>Stealth 4:</b> The host will regenerate missing organs over a long period of time.<br>\
+					  <b>Stage Speed 10:</b> The virus causes the host's internal organs to gain some self-correcting behaviour, preventing heart attacks and appendicitis.<br>"
+
+/datum/symptom/organ_restoration/severityset(datum/disease/advance/A)
+	. = ..()
+	if(A.stealth >= 4)
+		severity -= 1
+
+/datum/symptom/organ_restoration/Start(datum/disease/advance/A)
+	if(!..())
+		return
+	if(A.stage_rate >= 10)
+		curing = TRUE
+	if(A.stealth >= 4)
+		regenorgans = TRUE
+
+/datum/symptom/organ_restoration/Activate(datum/disease/advance/A)
+	if(!..())
+		return
+	var/mob/living/M = A.affected_mob
+	var/status = ORGAN_ORGANIC
+	if(MOB_ROBOTIC in A.infectable_biotypes)
+		status = null //if the disease is capable of interfacing with robotics, it is allowed to heal mechanical organs
+	if(A.stage >= 4)
+		M.adjustOrganLoss(ORGAN_SLOT_APPENDIX, -1, required_status = status)
+		M.adjustOrganLoss(ORGAN_SLOT_STOMACH, -1, required_status = status)
+		M.adjustOrganLoss(ORGAN_SLOT_LUNGS, -1, required_status = status)
+		M.adjustOrganLoss(ORGAN_SLOT_HEART, -1, required_status = status)
+		M.adjustOrganLoss(ORGAN_SLOT_LIVER, -1, required_status = status)
+		M.adjustOrganLoss(ORGAN_SLOT_TAIL, -1, required_status = status)
+		M.adjustOrganLoss(ORGAN_SLOT_WINGS, -1, required_status = status)
+		if(curing)
+			for(var/datum/disease/D in M.diseases)
+				if(istype(D, /datum/disease/appendicitis) || istype(D, /datum/disease/heart_failure))
+					D.cure()
+			if(iscarbon(M))
```

Comment:
You assume the mob is a carbon in other symptoms; if that's not a correct assumption, the other symptoms might runtime. If it is, this check is unnecessary?

Reply:
diseases are meant to be used on carbons, but symptoms are coded agnostically when it's possible. Futureproofing.

Reply:
That makes no sense considering only carbons can have organs, all this adds is a performance burden